packages/shared/src/components/post/block/PostTagsPanel.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { useBlockPost } from '../../../hooks/post/useBlockPost';
+import { Post } from '../../../graphql/posts';
+import { isNullOrUndefined } from '../../../lib/func';
+import { PostBlockedPanel } from './PostBlockedPanel';
+import CloseButton from '../../CloseButton';
+import { Button, ButtonSize } from '../../buttons/Button';
+import { SourceAvatar } from '../../profile/source';
+import useFeedSettings from '../../../hooks/useFeedSettings';
+import { BlockTagSelection } from './common';
+
+interface PostTagsPanelProps {
+  post: Post;
+  className?: string;
+  toastOnSuccess?: boolean;
+}
+
+export function PostTagsPanel({
+  post,
+  className,
+  toastOnSuccess = true,
+}: PostTagsPanelProps): ReactElement {
+  const { feedSettings } = useFeedSettings();
+  const [shouldBlockSource, setShouldBlockSource] = useState(
+    feedSettings?.excludeSources?.some(({ id }) => id === post.source.id),
+  );
+  const [tags, setTags] = useState<BlockTagSelection>(
+    feedSettings?.blockedTags?.reduce(
+      (block, tag) => ({ ...block, [tag]: true }),
+      {},
+    ),
```

Comment:
Since this is iteration using function to create initial value for `useState` is better so it does not run this `reduce` on each re-render.

Reply:
Same for `shouldBlockSource` above.

Reply:
Oh, nice catch. I've never know that.